src/test/java/uk/ac/cam/cl/dtg/segue/api/AdminFacadeTest.java

```diff
@@ -0,0 +1,423 @@
+import uk.ac.cam.cl.dtg.isaac.dos.AbstractUserPreferenceManager;
+import uk.ac.cam.cl.dtg.isaac.dos.users.Role;
+import uk.ac.cam.cl.dtg.isaac.dto.SegueErrorResponse;
+import uk.ac.cam.cl.dtg.isaac.dto.content.EmailTemplateDTO;
+import uk.ac.cam.cl.dtg.isaac.dto.users.RegisteredUserDTO;
+import uk.ac.cam.cl.dtg.segue.api.managers.IExternalAccountManager;
+import uk.ac.cam.cl.dtg.segue.api.managers.StatisticsManager;
+import uk.ac.cam.cl.dtg.segue.api.managers.UserAccountManager;
+import uk.ac.cam.cl.dtg.segue.api.monitors.IMisuseMonitor;
+import uk.ac.cam.cl.dtg.segue.auth.exceptions.NoUserException;
+import uk.ac.cam.cl.dtg.segue.auth.exceptions.NoUserLoggedInException;
+import uk.ac.cam.cl.dtg.segue.comm.EmailManager;
+import uk.ac.cam.cl.dtg.segue.comm.EmailType;
+import uk.ac.cam.cl.dtg.segue.dao.ILogManager;
+import uk.ac.cam.cl.dtg.segue.dao.SegueDatabaseException;
+import uk.ac.cam.cl.dtg.segue.dao.content.ContentManagerException;
+import uk.ac.cam.cl.dtg.segue.dao.content.GitContentManager;
+import uk.ac.cam.cl.dtg.segue.scheduler.SegueJobService;
+import uk.ac.cam.cl.dtg.util.PropertiesLoader;
+
+class AdminFacadeTest {
+  private PropertiesLoader properties;
+  private UserAccountManager userManager;
+  private EmailManager emailManager;
+  private GitContentManager contentManager;
+  private String contentIndex;
+  private ILogManager logManager;
+  private StatisticsManager statsManager;
+  private AbstractUserPreferenceManager userPreferenceManager;
+  private EventBookingManager eventBookingManager;
+  private SegueJobService segueJobService;
+  private IExternalAccountManager externalAccountManager;
+  private IMisuseMonitor misuseMonitor;
+  private AdminFacade adminFacade;
+
+  @BeforeEach
+  final void beforeEach() {
+    properties = createMock(PropertiesLoader.class);
+    userManager = createMock(UserAccountManager.class);
+    contentManager = createMock(GitContentManager.class);
+    contentIndex = "";
+    logManager = createMock(ILogManager.class);
+    statsManager = createMock(StatisticsManager.class);
+    userPreferenceManager = createMock(AbstractUserPreferenceManager.class);
+    eventBookingManager = createMock(EventBookingManager.class);
+    segueJobService = createMock(SegueJobService.class);
+    externalAccountManager = createMock(IExternalAccountManager.class);
+    misuseMonitor = createMock(IMisuseMonitor.class);
+    emailManager = createMock(EmailManager.class);
+    adminFacade = new AdminFacade(properties, userManager, contentManager, contentIndex, logManager, statsManager,
+        userPreferenceManager, eventBookingManager, segueJobService, externalAccountManager, misuseMonitor, emailManager);
+  }
+
+  @Nested
+  class ModifyUsersTeacherPendingStatus {
+    @Nested
+    class FailureResponses {
+      @Test
+      void modifyUsersTeacherPendingStatus_anonymousUser_returnsUnauthorised() throws NoUserLoggedInException {
+        HttpServletRequest mockRequest = replayMockServletRequest();
```

Comment:
You seem to be calling this before each test, so it sounds like a good candidate to be moved to `beforeEach`.